erpnext/stock/doctype/warehouse/warehouse.py

```diff
@@ -199,4 +229,54 @@ def get_warehouses_based_on_account(account, company=None):
 		frappe.throw(_("Warehouse not found against the account {0}")
 			.format(account))
 
-	return warehouses
\ No newline at end of file
+	return warehouses
+
+def get_warehouse_list(doctype, txt, filters, limit_start, limit_page_length=20, order_by="modified"):
+	"""Getting list of warehouse and there items details from Warehouse and Bin doctype."""
+	user = frappe.session.user
+	customers = []
+	docs = []
+	# Checking user have 'Warehouse' permission or not.
+	if frappe.has_permission("Warehouse"):
+		for doc in frappe.get_all("Contact", {"user": user}):
```

Comment:
```suggestion
		for doc in frappe.get_all("Contact", {"email_id": frappe.session.user}):
```
Just check whether frappe.session.user is email